controllers/agentcluster_controller.go

```diff
@@ -84,6 +85,11 @@ func (r *AgentClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	if paused := agentCluster.Annotations[capiv1.PausedAnnotation]; paused == "true" {
```

Comment:
It's there any `status.Conditions` in the `cluster` object to be updated once the AgentCluster is paused?, or should not be signalled once this situation happens?

Reply:
That's a good question, will look into it unless @eranco74 already knows?

Reply:
Looks like the `cluster` controller in the `cluster-api` repo already checks for it https://github.com/kubernetes-sigs/cluster-api/blob/59127b93260ad4a73e0a459a9ae7fc2e5eb08d0a/internal/controllers/cluster/cluster_controller_phases.go#L100